docs/src/main/asciidoc/reaugmentation.adoc

```diff
@@ -0,0 +1,66 @@
+////
+This guide is maintained in the main Quarkus repository
+and pull requests should be submitted there:
+https://github.com/quarkusio/quarkus/tree/main/docs/src/main/asciidoc
+////
+= Reaugment a Quarkus Application
+
+include::./attributes.adoc[]
+
+== What is augmentation?
+
+During augmentation the application is initialized - or let's say pre-initialized.
+Things that used to happen when an EAR file was deployed on a Jakarta-EE server such as parsing static configuration, creating proxy instances, etc. now happen at augmentation time.
+If you add CDI beans after augmentation they won't work (because of the missing proxy classes) and if you change build-time properties (e.g. `quarkus.datasource.db-kind`) after augmentation these changes will be ignored.
```

Comment:
```suggestion
CDI beans added after augmentation won't work (because of the missing proxy classes) as well as build-time properties (e.g. `quarkus.datasource.db-kind`) changed after augmentation will be ignored.
```